aiida/calculations/transfer.py

```diff
@@ -0,0 +1,250 @@
+    new remote folder saved in a RemoteData node).
+
+    Only files from the local computer and a single remote computer can be moved with a single
+    instance of this CalcJob (so you can't directly move files from remote to remote and all
+    RemoteData nodes must be from the same machine).
+
+    The user needs to provide three inputs:
+
+        * ``instructions``: a dict node specifying which files to copy from which nodes.
+        * ``source_nodes``: a dict of nodes, each with a unique identifier label as its key.
+        * ``metadata.computer``: the computer that contains the remote files and will contain
+          the final RemoteData node.
+
+    The ``instructions`` dict must have the ``retrieve_files`` flag. The CalcJob will create a
+    new folder in the remote machine (RemoteData) and put all the files there and will either:
+
+        (1) leave them there (``retrieve_files = False``) or ...
+        (2) create a local FolderData and copy there all the files (``retrieve_files = True``)
+
+    The `instructions` dict must also contain at least one list with specifications of which files
+    to copy and from where. All these lists take tuples of 3 that have the following format:
+
+    .. code-block:: python
+
+        [ ..., ( source_node_key, path_to_file_in_source, path_to_file_in_target), ... ]
+
+    where the ``source_node_key`` has to be the respective one used when providing the node in the
+    ``source_nodes`` input nodes dictionary.
+
+
+    The two main lists to include are ``local_files`` (for files to be taken from FolderData nodes)
+    and ``remote_files`` (for files to be taken from RemoteData nodes). Alternatively, files inside
+    of RemoteData nodes can instead be put in the ``symlink_files`` list: the only difference is that
+    files from the first list will be fully copied in the target RemoteData folder, whereas for the
+    files in second list only a symlink to the original file will be created there. This will only
+    affect the content of the final RemoteData target folder, but in both cases the full file will
+    be copied back in the local target FolderData (if ``retrieve_files = True``).
+    """
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+
+        spec.input(
+            'instructions',
+            valid_type=orm.Dict,
+            help='A dictionary containing the `retrieve_files` flag and at least one of the file lists:'
+            '`local_files`, `remote_files` and/or `symlink_files`.',
+            validator=validate_instructions,
+        )
+        spec.input_namespace(
+            'source_nodes',
+            valid_type=(orm.FolderData, orm.RemoteData),
+            dynamic=True,
+            help='All the nodes that contain files referenced in the instructions.',
+        )
+
+        # The transfer just needs a computer, the code are resources are set here
+        spec.inputs['code'].required = False
```

Comment:
if we are not going to use a code, maybe it is just better to get rid of the port altogether, i.e. `spec.inputs.pop('code', None)`

Reply:
I mean, since the original idea involved leaving this open for including in the future the posibility of performing the copy via a script (for cases like CSCS different storages, tapes, etc) then I'm not sure it makes sense to take it out if it will be eventualy be put back in (although now  I can't see specific cases, I guess it technically increases the chances of this later becoming a backwards incompatible change).

Reply:
I would see it from another point of view: if it is not used now whatsoever, and _if_ a code is specified, it will simply be ignored, I don't see the need to have it there, it can just create confusion. On top of that, if in the future we will add support, it can be added without breaking backwards compatibility. So I would really remove it